can/bus.py

```diff
@@ -67,12 +67,19 @@ def __init__(
     def __str__(self) -> str:
         return self.channel_info
 
-    def recv(self, timeout: Optional[float] = None) -> Optional[Message]:
+    def recv(
+        self,
+        timeout: Optional[float] = None,
+        channel: Optional[int] = None
+    ) -> Optional[Message]:
```

Comment:
Most of the benefit of python-can is the abstraction between the network interfaces via `Bus` and higher level code. Changing the `Bus` interface shouldn't be part of this PR.

Reply:
Then how would you go about changing the receive so that it is able to work on both channel 0 and channel 1 and not just channel 0?

Reply:
one other thing is the inability to specify the number of messages wanting to get received. with the canalyst-ii adapter there is an api function to obtain the number of messages in waiting and this can be used to create the array that is passed to the receive function of the array. the api will fill the array emptying the internal buffer on the device. The way it is currently done is inefficient and can lead to running out of available buffer.

Maybe it would be a better to add in \*\*kwargs as a parameter. this would allow any subclasses the flexibility to add additional parameters if they are needed.

Reply:
My preference is to keep the `**kwargs` for just the Bus initializer. Several interfaces add methods to `Bus` though, that would be okay. I'd suggest adding an internal bus state such as `recv_channel` and a setter method to modify it. Another method would also work for peeking at the number of messages in the recv buffer.

Is there support for receiving from all `Bus` channels which might make a better default in `_recv_internal`?

Reply:
it can be done by calling the API's receive method once for each channel. But this to will break API with `_recv_internal` because it can no longer return a message it would have to return a list or a tuple of messages.

I can override the recv method because the handling of the timeout is also not ideal because of the loop it makes and the repeated called to receive. The API already has a concession for supplying a timeout. This may be the best way to go about it and provide a channel parameter.

I had added it to the Bus class because I would have thought that there has to be more then just this one interface that has more then a single channel.

Let me make a change and see if it is something you like better.